charts-packages/ag-charts-community/src/scene/shape/range.ts

```diff
@@ -0,0 +1,128 @@
+import { Shape } from './shape';
+import { BBox } from '../bbox';
+import { RedrawType, SceneChangeDetection, RenderContext } from '../node';
+
+export class Range extends Shape {
+    static className = 'Range';
+
+    protected static defaultStyles = {
+        ...Shape.defaultStyles,
+        strokeWidth: 1,
+    };
+
+    constructor() {
+        super();
+        this.restoreOwnStyles();
+    }
+
+    private effectiveStrokeWidth: number = Range.defaultStyles.strokeWidth;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    x1: number = 0;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    y1: number = 0;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    x2: number = 0;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    y2: number = 0;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    startLine: boolean = false;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    endLine: boolean = false;
+
+    @SceneChangeDetection({ redraw: RedrawType.MINOR })
+    isRange: boolean = false;
+
+    computeBBox(): BBox {
+        return new BBox(this.x1, this.y1, this.x2 - this.x1, this.y2 - this.y1);
+    }
+
+    isPointInPath(_x: number, _y: number): boolean {
+        return false;
+    }
+
+    render(renderCtx: RenderContext) {
+        let { ctx, forceRender, stats } = renderCtx;
+
+        if (this.dirty === RedrawType.NONE && !forceRender) {
+            if (stats) stats.nodesSkipped += this.nodeCount.count;
+            return;
+        }
+
+        let x1 = this.x1;
+        let y1 = this.y1;
+        let x2 = this.x2;
+        let y2 = this.y2;
```

Comment:
Nit: Use a deconstruction here.

```suggestion
        let { x1, y1, x2, y2 } = this;
```